src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+  const dateParts = date.split(separator);
+  const newDate = ['', '', ''];
+
+  const formatMap = {
+    DD: 'день',
+    MM: 'місяць',
+    YYYY: 'рік',
+    YY: 'рік',
+  };
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    const format = fromFormat[i];
+    const partType = formatMap[format];
+    const partValue = dateParts[i];
+
+    let newIndex = -1;
+
+    for (let j = 0; j < toFormat.length; j++) {
+      if (formatMap[toFormat[j]] === partType) {
+        newIndex = j;
+        break;
+      }
+    }
+
+    if (format === 'YYYY' && toFormat[newIndex] === 'YY') {
```

Comment:
The conversion from 'YYYY' to 'YY' is correct, but it's worth noting that this will not work for years before 1000 or after 9999. If handling such dates is within the scope of your application, you'll need to adjust the logic.